src/System.Runtime.Intrinsics.Experimental/ref/System.Runtime.Intrinsics.cs

```diff
@@ -25,6 +25,148 @@ public struct Vector256<T> where T : struct
+        public static Vector128<T> CompareGreaterThan<T>(Vector128<T> left, Vector128<T> right) where T : struct { throw null; }
+        public static Vector64<T>  CompareGreaterThanZero<T>(Vector64<T>  value) where T : struct { throw null; }
+        public static Vector128<T> CompareGreaterThanZero<T>(Vector128<T> value) where T : struct { throw null; }
+        public static Vector64<T>  CompareGreaterThanOrEqual<T>(Vector64<T>  left, Vector64<T>    right) where T : struct { throw null; }
+        public static Vector128<T> CompareGreaterThanOrEqual<T>(Vector128<T> left, Vector128<T>   right) where T : struct { throw null; }
+        public static Vector64<T>  CompareGreaterThanOrEqualZero<T>(Vector64<T>  value) where T : struct { throw null; }
+        public static Vector128<T> CompareGreaterThanOrEqualZero<T>(Vector128<T> value) where T : struct { throw null; }
+        public static Vector64<T>  CompareLessThanZero<T>(Vector64<T>  value) where T : struct { throw null; }
+        public static Vector128<T> CompareLessThanZero<T>(Vector128<T> value) where T : struct { throw null; }
+        public static Vector64<T>  CompareLessThanOrEqualZero<T>(Vector64<T>  value) where T : struct { throw null; }
+        public static Vector128<T> CompareLessThanOrEqualZero<T>(Vector128<T> value) where T : struct { throw null; }
+        public static Vector64<T>  CompareTest<T>(Vector64<T>  left, Vector64<T>  right) where T : struct { throw null; }
+        public static Vector128<T> CompareTest<T>(Vector128<T> left, Vector128<T> right) where T : struct { throw null; }
+        public static Vector64<float>   Divide(Vector64<float>   left, Vector64<float>   right) { throw null; }
+        public static Vector128<float>  Divide(Vector128<float>  left, Vector128<float>  right) { throw null; }
+        public static Vector128<double> Divide(Vector128<double> left, Vector128<double> right) { throw null; }
+        public static T Extract<T>(Vector64<T>  vector, byte index) where T : struct { throw null; }
+        public static T Extract<T>(Vector128<T> vector, byte index) where T : struct { throw null; }
+        public static Vector64<T>  Insert<T>(Vector64<T>  vector, byte index, T data) where T : struct { throw null; }
+        public static Vector128<T> Insert<T>(Vector128<T> vector, byte index, T data) where T : struct { throw null; }
+        public static Vector64<sbyte>  LeadingSignCount(Vector64<sbyte>  value) { throw null; }
+        public static Vector64<short>  LeadingSignCount(Vector64<short>  value) { throw null; }
+        public static Vector64<int>    LeadingSignCount(Vector64<int>    value) { throw null; }
+        public static Vector128<sbyte> LeadingSignCount(Vector128<sbyte> value) { throw null; }
+        public static Vector128<short> LeadingSignCount(Vector128<short> value) { throw null; }
+        public static Vector128<int>   LeadingSignCount(Vector128<int>   value) { throw null; }
```

Comment:
Is there a reason why there is no `Vector128<long>` overload for this and the other methods that have `Vector128<long>`?  Do ARM processors not support that type?

We seem to only have 2 of them:

`public static Vector128<ulong>  Abs(Vector128<long>   value)`
`public static Vector128<long>   Negate(Vector128<long>   value)`

Reply:
A common limitations on Arm64 simd  instruction is that many do not support 64-bit elements.  If it is not supported,  The API must spell out exactly which primitives are supported.  

So if the intrinsic is generic and the vector is at least 128 bits long `long` and `ulong` are supported.

So `Add<T>`, `And<T>`, `AndNot<T>` ... all support `long` & `ulong` for `Vector128<T>`

`Vector64<T>` never supports 64 bit elements because it would not be a vector....

Reply:
I'm not sure I follow.

Why do we support using `long` in `Abs`, `Negate` , `Add`, `And`, `AndNot` ...., but we don't support `LeadingSignCount` with `long?

Reply:
> Because the processor explicitly doesn't support LeadingSignCount with long?

The processor does support LeadingSignCount with long in the base instruction set.
```
#if ARM64_HW_INTRINSIC_NYI
    public static class Base
    {
        public static bool IsSupported { get { throw null; } }

        public static uint  LeadingSignCount(int  value) { throw null; }
        public static ulong LeadingSignCount(long value) { throw null; }
        public static uint  LeadingZeroCount(uint  value) { throw null; }
        public static ulong LeadingZeroCount(ulong value) { throw null; }
    }
#endif
```

The processor doesn't support LeadingSignCount with Simd elements of long.

The Arm64 simd instructions `smin`, `smax`, `umin`, `umax`, `mul`, `cls`, `clz`  ... do not support an element size of 64 bits.  Apparently ARM decided cost/benefit was not strong enough to include in the ISA.  

We did not expose in the API, because they would not be intrinsics, but rather a sequence of instructions.